frontend/src/heat_map/democracy_viz.js

```diff
@@ -0,0 +1,351 @@
+        "max": 1
+    },
+    "v2elboycot": {
+        // what does ordinal values mean and how to convert it to
+        // what is in the database
+        "min": 0,
+        "max": 1
+    },
+    "v2lpname": { // this is a text value???
+        "min": 0,
+        "max": 1
+    },
+    "v2slpname": { // text value
+        "min": 0,
+        "max": 1
+    },
+    "v2tlpname": { //text value
+        "min": 0,
+        "max": 1
+    },
+    "v2psnatpar": { //ordinal
+        "min": 0,
+        "max": 1
+    },
+    "v2excrptps": { //ordinal
+        "min": 0,
+        "max": 1
+    },
+    "v2juaccnt": { //ordinal
+        "min": 0,
+        "max": 1
+    },
+    "v2svstterr": { //percentage
+        "min": 0,
+        "max": 100
+    },
+    "v2meharjrn": { //ordinal
+        "min": 0,
+        "max": 1
+    },
+    "v2peprisch": { //percentage
+        "min": 0,
+        "max": 100
+    },
+    "v2pesecsch": {
+        "min": 0,
+        "max": 100
+    },
+    "v2smonex": { //ordinal
+        "min": 0,
+        "max": 1
+    },
+    "e_migdppc": { //GDP so it can be anything
+        "min": 0,
+        "max": 60000
+    },
+    "e_mipopula": { //population
+        "min": 0,
+        "max": 100000
+    },
+    "gdpcapl": { //not in guidebook
+        "min": 0,
+        "max": 60000
+    },
+    "ethfrac": { //not in guidebook
+        "min": 0,
+        "max": 1
+    },
+    "relfrac": { //not in guidebook
+        "min": 0,
+        "max": 1
+    },
+    "numlang": { //not in guidebook
+        "min": 0,
+        "max": 1
+    },
+    "colbrit": { //not in guidebook
+        "min": 0,
+        "max": 1
+    },
+    "colfra": { //not in guidebook
+        "min": 0,
+        "max": 1
+    },
+    "gdpcap_currusd": { //not in guidebook
+        "min": 0,
+        "max": 1
+    },
+    "adultliteracy": { //not in guidebook
+        "min": 0,
+        "max": 1
+    },
+    "elecaccessrur": { //not in guidebook
+        "min": 0,
+        "max": 1
+    },
+};
+
+export class DemocracyMap extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            lng: 29,
+            lat: -22,
+            zoom: 4,
+            map_data: null,
+            mouseover_country: 'Nothing',
+        };
+        this.csrftoken = getCookie('csrftoken');
+        this.map_ref = React.createRef();
+        this.getCountryData = this.getCountryData.bind(this);
+    }
+
+    /**
+     * When this component is mounted to the DOM, get GeoJSON data from the server
+     */
+    async componentDidMount() {
+        try {
+            const res = await fetch('/api/africa_map_geojson/');
+            const geo_json = await res.json();
+            const map_data = this.project_features_and_create_svg_paths(geo_json);
+            this.setState({
+                map_data: map_data,
+            });
+        } catch (e) {
+            console.log(e);
+        }
+    }
+
+    project_features_and_create_svg_paths(geo_json) {
```

Comment:
also copy/paste from map quiz -- please factor this common mapping functionality out.